rbs/MethodTypeTranslator.cc

```diff
@@ -0,0 +1,307 @@
+            if (auto e = ctx.beginError(docLoc, core::errors::Rewriter::RBSError)) {
+                e.setHeader("Unexpected node type `{}` in keyword argument value, expected `{}`",
+                            rbs_node_type_name(hash_node->value), "FunctionParam");
+            }
+            continue;
+        }
+
+        rbs_ast_symbol_t *keyNode = (rbs_ast_symbol_t *)hash_node->key;
+        rbs_types_function_param_t *valueNode = (rbs_types_function_param_t *)hash_node->value;
+        auto arg = RBSArg{docLoc, keyNode, valueNode->type, optional};
+        args.emplace_back(arg);
+    }
+}
+
+} // namespace
+
+sorbet::ast::ExpressionPtr MethodTypeTranslator::methodSignature(core::MutableContext ctx,
+                                                                 sorbet::ast::MethodDef *methodDef,
+                                                                 MethodType methodType,
+                                                                 std::vector<Comment> annotations) {
+    rbs_methodtype_t *node = methodType.node.get();
+
+    if (node->type->type != RBS_TYPES_FUNCTION) {
+        auto errLoc = TypeTranslator::nodeLoc(methodType.loc, node->type);
+        if (auto e = ctx.beginError(errLoc, core::errors::Rewriter::RBSError)) {
+            e.setHeader("Unexpected node type `{}` in method signature, expected `{}`", rbs_node_type_name(node->type),
+                        "Function");
+        }
+
+        return ast::MK::Untyped(methodType.loc);
+    }
+
+    // Collect type parameters
+
+    std::vector<std::pair<core::LocOffsets, core::NameRef>> typeParams;
+    for (rbs_node_list_node_t *list_node = node->type_params->head; list_node != nullptr; list_node = list_node->next) {
+        auto loc = TypeTranslator::nodeLoc(methodType.loc, list_node->node);
+
+        if (list_node->node->type != RBS_AST_TYPEPARAM) {
+            if (auto e = ctx.beginError(loc, core::errors::Rewriter::RBSError)) {
+                e.setHeader("Unexpected node type `{}` in type parameter list, expected `{}`",
+                            rbs_node_type_name(list_node->node), "TypeParam");
+            }
+
+            continue;
+        }
+
+        auto node = (rbs_ast_typeparam_t *)list_node->node;
+        rbs_constant_t *constant = rbs_constant_pool_id_to_constant(fake_constant_pool, node->name->constant_id);
+        std::string string(constant->start, constant->length);
+        typeParams.emplace_back(loc, ctx.state.enterNameUTF8(string));
+    }
+
+    // Collect positionals
+
+    rbs_types_function_t *functionType = (rbs_types_function_t *)node->type;
+    std::vector<RBSArg> args;
+
+    collectArgs(ctx, methodType.loc, functionType->required_positionals, args, false);
+
+    rbs_node_list_t *optionalPositionals = functionType->optional_positionals;
+    if (optionalPositionals && optionalPositionals->length > 0) {
+        collectArgs(ctx, methodType.loc, optionalPositionals, args, false);
```

Comment:
Is the second call really supposed to use `false` for `optional`?  If so, we never pass `true` to `collectArgs`, so maybe we can delete the `optional` argument?  If it's supposed to use `true`, can we add test coverage that would have caught this case?